src/dao/DAO.js

```diff
@@ -252,38 +275,51 @@ class DAO<TEntity: { id: EntityID }, TEntityMutator> extends BaseDAO<
 
   waitForLoaded<TResponse>(
     fn: () => LoadObject<TResponse>,
-    timeout: number = 10000
+    timeout: number = 10000,
   ): Promise<TResponse> {
-    return new Promise((resolve, reject) => {
-      const fetchAndResolve = () => {
-        const loader = fn().map((result) => {
-          if (!Array.isArray(result)) {
+    return new Promise(
+      (
+        resolve: (response: TResponse) => void,
+        reject: (error: Error) => void,
+      ) => {
+        setTimeout((): void => reject(new Error('Timeout!')), timeout);
+
+        const fetchAndResolve = () => {
+          const loader = fn().map((result: $FlowFixMe): $FlowFixMe => {
```

Comment:
sorry, but I have no idea how to make Flow happy here, hold for nextTime

Reply:
Yeah... this could be nasty